internal/benchmarks/src/transaction-results.bench.ts

```diff
@@ -55,45 +69,55 @@ describe('Transaction Submission Benchmarks', () => {
     const { transaction } = await pendingTx.waitForResult();
 
     expect(transaction).toBeDefined();
+  };
+
+  bench('should successfully transfer a single asset between wallets 10 times', async () => {
+    for (let i = 0; i < 10; i++) {
+      await transfer();
+    }
   });
 
-  bench('should successfully perform a batch transfer', async () => {
-    const receiver1 = Wallet.generate({ provider });
-    const receiver2 = Wallet.generate({ provider });
-    const receiver3 = Wallet.generate({ provider });
+  bench('should successfully conduct a custom transfer between wallets 10 times', async () => {
+    for (let i = 0; i < 10; i++) {
+      await customTransfer();
+    }
+  });
 
-    const amountToTransfer1 = 989;
-    const amountToTransfer2 = 699;
-    const amountToTransfer3 = 122;
+  bench('should successfully perform a batch transfer 10 times', async () => {
+    for (let i = 0; i < 10; i++) {
+      const amountToTransfer1 = 989;
+      const amountToTransfer2 = 699;
+      const amountToTransfer3 = 122;
 
-    const transferParams: TransferParams[] = [
-      {
-        destination: receiver1.address,
-        amount: amountToTransfer1,
-        assetId: provider.getBaseAssetId(),
-      },
-      { destination: receiver2.address, amount: amountToTransfer2, assetId: TestAssetId.A.value },
-      { destination: receiver3.address, amount: amountToTransfer3, assetId: TestAssetId.B.value },
-    ];
+      const transferParams: TransferParams[] = [
+        {
+          destination: receiver1.address,
+          amount: amountToTransfer1,
+          assetId: provider.getBaseAssetId(),
+        },
+        { destination: receiver2.address, amount: amountToTransfer2, assetId: TestAssetId.A.value },
+        { destination: receiver3.address, amount: amountToTransfer3, assetId: TestAssetId.B.value },
+      ];
 
-    const tx = await wallet.batchTransfer(transferParams);
+      const tx = await wallet.batchTransfer(transferParams);
```

Comment:
@maschad It seems one of the [failing tests](https://github.com/FuelLabs/fuels-ts/actions/runs/10894541904/job/30231421374#step:5:143) is triggered by this `batchTransfer` call.

I just double-checked the wallet pvt key and updated the GH secret to be sure, then re-ran the workflow, and it failed like before.

@Torres-ssf Thoughts?

Reply:
@arboleya that's strange because based on the [CI logs](https://github.com/FuelLabs/fuels-ts/actions/runs/10894541904/job/30257363622?pr=3166#step:5:113) it looks like it's failing at the [contract read call](https://github.com/FuelLabs/fuels-ts/pull/3166/files#diff-31f4ad5f8775c72e8c13a0bc07d99b7dba56c4bf9914d6af75eab3a146689f3fR63)

Reply:
Oh, right, the first error changed between the two runs. I linked the first, and you linked the second.

I was referring to the second error on the bottom rectangles.
![runs-diff](https://github.com/user-attachments/assets/47aabe01-a30d-4e18-8851-1b89a3a20cea)

Reply:
@arboleya @maschad, I am unsure about what is happening here. I will take a closer look

Reply:
I've resolved it. @Torres-ssf did some digging and he discovered that Codspeed actually runs our benchmarks twice in it's own environment (independent of any loop) as a result looping this 10 times in quick succession seems to be problematic on devnet - perhaps UTXO caching? Not sure.